### PR TITLE
HW-26

### PR DESCRIPTION
**Pull request by andromash**

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/IndexController.java [hunk 1]
```diff
@@ -2,10 +2,12 @@
...
 import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+@WebServlet(urlPatterns = {"/", "/index"})
 public class IndexController extends HttpServlet {
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
```
Comment (andriylagun) [inline src/main/java/com/internet/shop/controllers/IndexController.java]: do we need 2 paths for homepage?

**Review by andriylagun — COMMENTED** [review]
Good JOB!

**Review by andriylagun — APPROVED** [review]

File: src/main/java/com/internet/shop/controllers/IndexController.java (hunk 1 above)
Comment (andromash) [reply src/main/java/com/internet/shop/controllers/IndexController.java]: we use both variants, so yes

File: src/main/java/com/internet/shop/controllers/AddToCartController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import com.internet.shop.service.ProductService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(name = "addToCart", urlPatterns = "/buyProduct")
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/AddToCartController.java]: did you read checklist? 
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

Comment (andromash) [reply src/main/java/com/internet/shop/controllers/AddToCartController.java]: sorry, missed that part

File: src/main/java/com/internet/shop/controllers/AddToCartController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import com.internet.shop.service.ProductService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(name = "addToCart", urlPatterns = "/product/buy")
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/AddToCartController.java]: should we know the name of the servlet?

File: src/main/java/com/internet/shop/controllers/CurrentCartController.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(name = "currentCart", urlPatterns = "/current/cart")
+public class CurrentCartController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/CurrentCartController.java]:
```suggestion
public class GetCurrentCartController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/ProductsController.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(name = "products", urlPatterns = "/product/all")
+public class ProductsController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/ProductsController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/UsersController.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(name = "allUsers", urlPatterns = "/user/all")
+public class UsersController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/UsersController.java]:
```suggestion
public class GetAllUsersController extends HttpServlet {
```

**Review by Sofasmile — COMMENTED** [review]
how about:
Реализовать добавление товара?

File: src/main/java/com/internet/shop/controllers/AddNewProductController.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.math.BigDecimal;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/product/new")
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/AddNewProductController.java]:
```suggestion
@WebServlet("/product/add")
```

File: src/main/java/com/internet/shop/controllers/AddNewProductController.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        String name = req.getParameter("name");
+        String price = req.getParameter("price");
+        if (price.matches("^\\d+(\\.\\d+)*$")) {
+            productService.create(new Product(name, new BigDecimal(price)));
+            resp.sendRedirect(req.getContextPath() + "/index");
+        } else {
+            req.setAttribute("message", "You entered invalid price format");
+            req.getRequestDispatcher("/WEB-INF/views/product/add.jsp").forward(req, resp);
+        }
+
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/AddNewProductController.java]: remove one empty line

File: src/main/java/com/internet/shop/controllers/GetCurrentCartController.java [hunk 9]
```diff
@@ -12,8 +12,8 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-@WebServlet(name = "currentCart", urlPatterns = "/current/cart")
-public class CurrentCartController extends HttpServlet {
+@WebServlet("/current/cart")
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/GetCurrentCartController.java]: maybe better: /cart/products
what do you think?

File: src/main/java/com/internet/shop/controllers/DeleteUserController.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+@WebServlet("/user/delete")
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long userId = Long.valueOf(req.getParameter("userId"));
+        userService.delete(userId);
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/DeleteUserController.java:24]: how about removing the shopping cart here?

**Review by Sofasmile — COMMENTED** [review]
the last comment))

File: src/main/java/com/internet/shop/controllers/DeleteUserController.java (hunk 10 above)
Comment (andromash) [reply src/main/java/com/internet/shop/controllers/DeleteUserController.java:24]: my eyes are already wet 💯

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**